IdSSLOpenSSLHeaders.pas

```diff
@@ -22709,6 +22728,13 @@ function LoadSSLCryptoLibrary: TIdLibHandle;
   {$IFDEF WINDOWS}
   //On Windows, you should use SafeLoadLibrary because
   //the LoadLibrary API call messes with the FPU control word.
+  for i := 0 to SSLCLIBS_LAST do begin
+    Result := SafeLoadLibrary(GIdOpenSSLPath + SSLCLIBS[i]);
+    if Result <> IdNilHandle then begin
+      Exit;
+    end;
+  end;
+    
   Result := SafeLoadLibrary(GIdOpenSSLPath + SSLCLIB_DLL_name);

```

Comment:
Since you included `SSLCLIB_DLL_name` in the `SSLCLIBS` array, this load after the loop exits is redundant and should be removed.